src/libraries/ctrlLib/include/iCub/ctrl/filters.h

```diff
@@ -42,12 +42,20 @@ namespace iCub
 namespace ctrl
 {
 
+class IFilter {
+public:
+    virtual ~IFilter() = default;
```

Comment:
C++11 is not currently enabled for icub-main, I suggest implementing the destructor directly.